projects/js-packages/charts/src/components/line-chart/line-chart.tsx

```diff
@@ -17,11 +17,31 @@ import { withResponsive } from '../shared/with-responsive';
 import styles from './line-chart.module.scss';
 import type { BaseChartProps, DataPointDate, SeriesData } from '../../types';
 
+type CurveType = 'smooth' | 'linear' | 'monotone';
+
 const X_TICK_WIDTH = 100;
 
+const getCurveType = ( type?: CurveType, smoothing?: boolean ) => {
+	// First check for explicit curve type
+	if ( type ) {
+		switch ( type ) {
+			case 'smooth':
+				return curveCatmullRom;
+			case 'monotone':
+				return curveMonotoneX;
+			case 'linear':
+				return curveLinear;
+		}
+	}
+
+	// Fall back to legacy smoothing prop
+	return smoothing ? curveCatmullRom : curveLinear;
+};
```

Comment:
I would test on `smoothing` at the very beginning of the function, so that if the option is off, it could return early and doesn't have to run the logic below it.

Reply:
OK good point. Let's see if we can get rid of smoothing first. That would make this much more straight forward to implement 😄